### PR TITLE
insights: Don't return an error when checking for default mode on a standard query

### PR DESCRIPTION
**Pull request by CristinaBirkel**
## Description

When getting a default aggregation mode, `canAggregateByCaptureGroup` is calling `NewPatternReplacer` to check to see if there are any capture groups. `NewPatternReplacer` returns errors if it isn't a `regexp`, and we don't want to throw an error on that if we're just using it to check for capture groups.

So the solution here is to check specifically for that error, and to return `false` instead of an error when it's hit.

## Test plan

First, run a standard query that will check for the default mode. Example:
```
{
  searchQueryAggregate(query: "graphql.NewSchema", patternType: standard) {
    aggregations(limit: 10) {
      __typename
      ... on ExhaustiveSearchAggregationResult {
        groups {
          label
          count
          query
        }
        otherResultCount
        otherGroupCount
      }
      ... on SearchAggregationNotAvailable {
        reason
      }
    }
  }
}
```
Verify that you get a "pattern parsing" error. Then try again with this fix in place and verify that it works.

### CODE REVIEW

File: enterprise/internal/insights/resolvers/aggregates_resolvers.go [hunk 1]
```diff
@@ -338,6 +338,11 @@ func canAggregateByCaptureGroup(searchQuery, patternType string) (bool, error) {
 	// A query should contain at least a regexp pattern and capture group to allow capture group aggregation.
 	// Only the first capture group will be used for aggregation.
 	replacer, err := querybuilder.NewPatternReplacer(querybuilder.BasicQuery(searchQuery), searchType)
+	// If this error is returned, it means there are no capture groups.
+	if err == querybuilder.UnsupportedPatternTypeErr {
```
Comment (leonore) [inline enterprise/internal/insights/resolvers/aggregates_resolvers.go:342]: shouldn't this also check on multiple patterns?

Comment (CristinaBirkel) [reply enterprise/internal/insights/resolvers/aggregates_resolvers.go:342]: I had that originally, but I think that multiple patterns is the `and`/`or` case. Oh maybe I didn't realize that we are supporting `and`/`or` in other types of aggregations besides capture groups. I'll add it back in.

Comment (CristinaBirkel) [reply enterprise/internal/insights/resolvers/aggregates_resolvers.go]: Oh actually there is that test case for it; I think I'm confused. Do we support `and`/`or` for non-capture-group aggregations? It seems to work okay for me when I try it out in the UI.

Comment (leonore) [reply enterprise/internal/insights/resolvers/aggregates_resolvers.go]: I think we do, because we support it for normal insights.

**Review by leonore — APPROVED** [review]
tested this out locally and fixed/added unit tests

will land to unblock testing

**Pull request merged**